tests/Collection/ArrayCollectionHelperTest.php

```diff
@@ -0,0 +1,363 @@
+            'options' => [
+                'unittest' => true,
+            ],
+//            'features' => new ReplaceValue(
+//                [
+//                    'gii',
+//                ]
+//            ),
+        ]))->addModifier(
+            (new ReplaceValue())
+                ->forKey('features')
+                ->toValue(['gii'])
+        );
+
+        $result = ArrayCollectionHelper::merge($a, $b)->toArray();
+        $expected = [
+            'name' => 'Yii',
+            'version' => '1.1',
+            'options' => [
+                'namespace' => false,
+                'unittest' => true,
+            ],
+            'features' => [
+                'gii',
+            ],
+        ];
+
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMergeWithRemoveKeys(): void
+    {
+        $a = [
+            'name' => 'Yii',
+            'version' => '1.0',
+        ];
+        $b = (new ArrayCollection([
+            'version' => '1.1',
+            'options' => [],
+//            new RemoveKeys(),
+        ]))->addModifier(
+            new RemoveKeys()
+        );
+
+        $result = ArrayCollectionHelper::merge($a, $b)->toArray();
+        $expected = [
+            'Yii',
+            '1.1',
+            [],
+        ];
+
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMergeWithReverseBlock(): void
+    {
+        $a = [
+            'name' => 'Yii',
+            'options' => [
+                'option1' => 'valueA',
+                'option3' => 'valueAA',
+            ],
+            'version' => '1.0',
+            'meta' => ['a' => 1],
+        ];
+        $b = (new ArrayCollection([
+            'version' => '1.1',
+            'options' => [
+                'option1' => 'valueB',
+                'option2' => 'valueBB',
+            ],
+//            ReverseBlockMerge::class => new ReverseBlockMerge(),
+        ]))->addModifier(
+            new MergeWithKeysAsReverseMerge(),
+        );
+
+        $result = ArrayCollectionHelper::merge($a, $b)->toArray();
+        $expected = [
+            'version' => '1.1',
+            'options' => [
+                'option1' => 'valueB',
+                'option2' => 'valueBB',
+                'option3' => 'valueAA',
+            ],
+            'name' => 'Yii',
+            'meta' => ['a' => 1],
+        ];
+
+        $this->assertSame($expected, $result);
+    }
+
+    public function testMergeSimpleArrayWithReverseBlock(): void
+    {
+        $a = [
+            'A',
+            'B',
+        ];
+        $b = (new ArrayCollection([
+            'C',
+            'B',
+          //  ReverseBlockMerge::class => new ReverseBlockMerge(),
+        ]))->addModifier(
+            new MergeWithKeysAsReverseMerge()
+        );
+
+        $this->assertSame(['C', 'B', 'A'], ArrayCollectionHelper::merge($a, $b)->toArray());
+    }
+
+//    public function testMergeWithAlmostReverseBlock(): void
```

Comment:
?

Reply:
This file with tests is copy of current array modifiers test adpated to collections.
I commented old modifiers for comparsions was / became.